tests/functional/tests/engine/test_io_zero_size.py

```diff
@@ -0,0 +1,45 @@
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+from ctypes import c_int
+
+from pyocf.types.cache import Cache
+from pyocf.types.core import Core
+from pyocf.types.volume import Volume
+from pyocf.types.data import Data
+from pyocf.types.io import IoDir
+from pyocf.utils import Size
+from pyocf.types.shared import OcfCompletion
+
+
+def test_io_zero_size(pyocf_ctx):
+    cache_device = Volume(Size.from_MiB(30))
+    core_device = Volume(Size.from_MiB(60))
+
+    cache = Cache.start_on_device(cache_device)
+    core = Core.using_device(core_device)
+
+    cache.add_core(core)
+
+    cache_device.reset_stats()
+    core_device.reset_stats()
+
+    io = core.new_core_io(cache.get_default_queue(), Size.from_sector(1).B,
+                          0, IoDir.WRITE, 0, 0)
+    io.set_data(0)
+
+    cmpl = OcfCompletion([("err", c_int)])
+    io.callback = cmpl.callback
+
+    io.submit()
+    cmpl.wait()
+
+    assert cmpl.results["err"] == -5  # I/O error
```

Comment:
Does this test pass? Because in this scenario CAS should return `-OCF_ERR_INVAL` which has numerical value of -1000000, not -5.

Reply:
Should, but this code's error has value of -5, which is I/O error in `errno-base.h`. See here: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/asm-generic/errno-base.h
I don't know why, because it's not returned anywhere in OCF's code. I'm checking it.

Reply:
`OCF_ERR_INVAL` is defined in [https://github.com/Open-CAS/ocf/blob/master/inc/ocf_err.h](https://github.com/Open-CAS/ocf/blob/master/inc/ocf_err.h) and definitely it doesn't have value -5.

Reply:
The test was written badly, now it works and the error code is correct.